kuma/wiki/content.py

```diff
@@ -559,6 +565,27 @@ def add_to_attr(attr_name, add_list):
             yield token
 
 
+class PreNotranslateClassFilter(html5lib_Filter):
+    """
+    Filter which annotates <pre> tags with the 'notranslate' class.
+    """
+
+    def __iter__(self):
+        input = html5lib_Filter.__iter__(self)
+        for token in input:
+            if token["type"] == "StartTag" and token["name"] == "pre":
+                attrs = dict(token["data"]) or {(None, "class"): ""}
+                for (namespace, name), value in attrs.copy().items():
+                    if name == "class" and "notranslate" not in value.lower().split():
```

Comment:
I was worried about the use of `.lower()` here, since I wasn't sure if something like `class="NOTRANSLATE"` would work the same as `class="notranslate"`, so I checked this using the Google Translate Chrome extension. It turns out that `class="NOTRANSLATE"` and `class="noTranslate"` do **not** work, but interestingly `class="NoTranslate"` does. Your call on whether we should also check for `NoTranslate`, but I think we should remove `.lower()` here:
```suggestion
                    if name == "class" and "notranslate" not in value.split():
```

Reply:
Good catch!
Also, the chances that someone already put in `<pre class="NOTransLate">` are extremely rare. Let's do your suggestion.